### PR TITLE
feat: desktop i18n

### PR DESCRIPTION
**Pull request by zjy365**
### Summary
🌐🚀🐛

This pull request adds internationalization support to the frontend desktop app using the `react-i18next` library. It updates several components to use the `useTranslation` hook and replaces the static UI texts with the `t` function. It also adds the English and Chinese translations in the `./frontend/desktop/public/locales` folder. Additionally, it improves the performance by importing some components dynamically and removing a redundant mask layer. It also changes the `debug` option of the `i18next` initialization to depend on the `NODE_ENV` variable.

> _The desktop app needed some changes_
> _To support different languages_
> _With `useTranslation` hook_
> _And some `translation.json` book_
> _The UI texts got new ranges_

### Walkthrough
*  Replace the English and Chinese translation files with new versions that contain the UI texts ([link](https://github.com/labring/sealos/pull/3221/files?diff=unified&w=0#diff-469b9b5a78e15a3f4d47da2694d3dd87938abdcafc7df2e7e5c37c32088638bdL2-R12), [link](https://github.com/labring/sealos/pull/3221/files?diff=unified&w=0#diff-9360e8b98a4d6bc0e4f5b098d0d4cbd72e91f38400c3cb145f70d69d20be4f28L2-R12))
*  Import the `useTranslation` hook from the `react-i18next` library in several components to enable internationalization of the UI texts ([link](https://github.com/labring/sealos/pull/3221/files?diff=unified&w=0#diff-f080c4cc5f58f00e0d58040ad4f90c0bdf6b500e0ea8d8711bacaa900803500bL11-R15), [link](https://github.com/labring/sealos/pull/3221/files?diff=unified&w=0#diff-6eec7bb4fac557f6da1a4c82880637cbb656e3b08ab0b2adcc12d6d9777aa028L18-R23), [link](https://github.com/labring/sealos/pull/3221/files?diff=unified&w=0#diff-3a46b3c3bb2bbb0c0372478d6355a28810f9ca0ebbd32b49115db7b90190d9c4R4), F8L7L7, [link](https://github.com/labring/sealos/pull/3221/files?diff=unified&w=0#diff-322ecb0d4c3fbb372838a948e5d24b716b5ec69c8b50fcf308b74fba2f468bb3L5-R8), [link](https://github.com/labring/sealos/pull/3221/files?diff=unified&w=0#diff-05ab50df40ed287b38184d4cffd7b53e8f144ab5af25b1a388fe5569aaef2ec3R9), [link](https://github.com/labring/sealos/pull/3221/files?diff=unified&w=0#diff-ddb789991f340d8a4490e222f156d84cb536f5b343a6038bdc42cdc7519affadR8))
*  Wrap the UI texts with the `t` function from the `useTranslation` hook to translate them to the current language in several components ([link](https://github.com/labring/sealos/pull/3221/files?diff=unified&w=0#diff-f080c4cc5f58f00e0d58040ad4f90c0bdf6b500e0ea8d8711bacaa900803500bL64-R72), [link](https://github.com/labring/sealos/pull/3221/files?diff=unified&w=0#diff-6eec7bb4fac557f6da1a4c82880637cbb656e3b08ab0b2adcc12d6d9777aa028L118-R122), [link](https://github.com/labring/sealos/pull/3221/files?diff=unified&w=0#diff-3a46b3c3bb2bbb0c0372478d6355a28810f9ca0ebbd32b49115db7b90190d9c4L46-R57), [link](https://github.com/labring/sealos/pull/3221/files?diff=unified&w=0#diff-942cf93ff546b74643d32be5bdb582b156157a33ace9bb5bf07ca080c1336c60L43-R45), [link](https://github.com/labring/sealos/pull/3221/files?diff=unified&w=0#diff-942cf93ff546b74643d32be5bdb582b156157a33ace9bb5bf07ca080c1336c60L117-R119), [link](https://github.com/labring/sealos/pull/3221/files?diff=unified&w=0#diff-322ecb0d4c3fbb372838a948e5d24b716b5ec69c8b50fcf308b74fba2f468bb3L29-R31), [link](https://github.com/labring/sealos/pull/3221/files?diff=unified&w=0#diff-05ab50df40ed287b38184d4cffd7b53e8f144ab5af25b1a388fe5569aaef2ec3L111-R113), [link](https://github.com/labring/sealos/pull/3221/files?diff=unified&w=0#diff-05ab50df40ed287b38184d4cffd7b53e8f144ab5af25b1a388fe5569aaef2ec3L120-R122), [link](https://github.com/labring/sealos/pull/3221/files?diff=unified&w=0#diff-05ab50df40ed287b38184d4cffd7b53e8f144ab5af25b1a388fe5569aaef2ec3L127-R129), [link](https://github.com/labring/sealos/pull/3221/files?diff=unified&w=0#diff-05ab50df40ed287b38184d4cffd7b53e8f144ab5af25b1a388fe5569aaef2ec3L135-R137), [link](https://github.com/labring/sealos/pull/3221/files?diff=unified&w=0#diff-05ab50df40ed287b38184d4cffd7b53e8f144ab5af25b1a388fe5569aaef2ec3L157-R162), [link](https://github.com/labring/sealos/pull/3221/files?diff=unified&w=0#diff-05ab50df40ed287b38184d4cffd7b53e8f144ab5af25b1a388fe5569aaef2ec3L182-R188))
*  Declare the `t` and `i18n` variables from the `useTranslation` hook to access the translation function and the current language in some components ([link](https://github.com/labring/sealos/pull/3221/files?diff=unified&w=0#diff-3a46b3c3bb2bbb0c0372478d6355a28810f9ca0ebbd32b49115db7b90190d9c4R18-R19), [link](https://github.com/labring/sealos/pull/3221/files?diff=unified&w=0#diff-ddb789991f340d8a4490e222f156d84cb536f5b343a6038bdc42cdc7519affadR15))
*  Add a conditional rendering logic to display the date and day in different formats depending on the current language in the `time.tsx` component ([link](https://github.com/labring/sealos/pull/3221/files?diff=unified&w=0#diff-3a46b3c3bb2bbb0c0372478d6355a28810f9ca0ebbd32b49115db7b90190d9c4L46-R57))
*  Add a commented out code block to show an example of how to switch the language using the `i18n.changeLanguage` function in the `user_menu/index.tsx` component ([link](https://github.com/labring/sealos/pull/3221/files?diff=unified&w=0#diff-ddb789991f340d8a4490e222f156d84cb536f5b343a6038bdc42cdc7519affadR26-R40))
*  Change the `debug` option of the `i18next` initialization to depend on the `NODE_ENV` environment variable in the `i18n.ts` file to enable debugging only in the development mode ([link](https://github.com/labring/sealos/pull/3221/files?diff=unified&w=0#diff-81d38b13b7ca2783a1c9d8560b95a96ea5eba635c79fb0cf8e910444c974c59eL19-R19))
*  Import the `MoreButton` and `MoreApps` components dynamically using the `next/dynamic` library in the `desktop_content/index.tsx` and `layout/index.tsx` files to improve the performance and avoid server-side rendering issues ([link](https://github.com/labring/sealos/pull/3221/files?diff=unified&w=0#diff-6eec7bb4fac557f6da1a4c82880637cbb656e3b08ab0b2adcc12d6d9777aa028L2), [link](https://github.com/labring/sealos/pull/3221/files?diff=unified&w=0#diff-6eec7bb4fac557f6da1a4c82880637cbb656e3b08ab0b2adcc12d6d9777aa028R10), [link](https://github.com/labring/sealos/pull/3221/files?diff=unified&w=0#diff-4ae759586b8ce6e86c0d3949c7e588f4dcfd84aa3232cf567704c324c85196abL8-R11))
*  Add a `div` element with the `styles.appMask` class name to create a mask layer for the app window when it is not in the highest layer in the `app_window/index.tsx` file ([link](https://github.com/labring/sealos/pull/3221/files?diff=unified&w=0#diff-07920ae91c024b06d398abdc6a907da7a5d7316784fba23d4450241bc0bf214dL185-R192))
*  Remove a duplicated `div` element with the `styles.appMask` class name from the `app_window/index.tsx` file ([link](https://github.com/labring/sealos/pull/3221/files?diff=unified&w=0#diff-07920ae91c024b06d398abdc6a907da7a5d7316784fba23d4450241bc0bf214dL198-L205))
*  Delete the `frontend/desktop/public/locales/zh-CN/translation.json` file ([link](https://github.com/labring/sealos/pull/3221/files?diff=unified&w=0#diff-fb857733bc708fa5c48c2ef8392670a52b78ce1ee931aeefe79e093189808297))

### CODE REVIEW

File: frontend/desktop/public/locales/en/translation.json [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-    "ValuationStandard": "Standard"
-  },
-  "Valuation": {
-    "Standard": "ValuationStandard",
-    "Name": "Name",
-    "Unit": "Unit",
-    "Price": "Price"
-  }
+  "Cost Center": "Cost Center",
+  "Laf on SealOS": "Laf on SealOS",
```
Comment (fanux) [inline frontend/desktop/public/locales/en/translation.json:3]: SealOS->Sealos

File: frontend/desktop/public/locales/en/translation.json [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
+  "App Launchpad": "App Launchpad",
+  "Sealos Document": "Sealos Document",
+  "Terminal": "Terminal",
+  "more apps": "more apps",
```
Comment (fanux) [inline frontend/desktop/public/locales/en/translation.json:7]: More Apps

File: frontend/desktop/public/locales/en/translation.json [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
+  "消息中心": "Message Center",
```
Comment (fanux) [inline frontend/desktop/public/locales/en/translation.json:8]: "key" should be uniformly written in English.

File: frontend/desktop/public/locales/zh/translation.json [hunk 4]
```diff
@@ -1,22 +1,14 @@
...
-    "ValuationStandard": "计价标准"
-  },
-  "Valuation": {
-    "Standard": "计价标准",
-    "Name": "资源名",
-    "Unit": "单位",
-    "Price": "价格"
-  }
+  "Cost Center": "费用中心",
+  "Laf on SealOS": "Laf on SealOS",
```
Comment (fanux) [inline frontend/desktop/public/locales/zh/translation.json:3]: 函数计算

File: frontend/desktop/public/locales/zh/translation.json [hunk 5]
```diff
@@ -1,22 +1,14 @@
...
+  "App Launchpad": "应用中心",
+  "Sealos Document": "文档中心",
+  "Terminal": "终端",
+  "more apps": "更多应用",
+  "消息中心": "消息中心",
```
Comment (fanux) [inline frontend/desktop/public/locales/zh/translation.json:8]: "key" should be uniformly written in English.

File: frontend/desktop/public/locales/zh/translation.json [hunk 6]
```diff
@@ -1,22 +1,14 @@
...
-  },
-  "Valuation": {
-    "Standard": "计价标准",
-    "Name": "资源名",
-    "Unit": "单位",
-    "Price": "价格"
-  }
+  "Cost Center": "费用中心",
+  "Laf on Sealos": "函数计算",
+  "App Launchpad": "应用中心",
```
Comment (zzjin) [inline frontend/desktop/public/locales/zh/translation.json:4]: Apps should keep translations inside CR?

File: frontend/desktop/Makefile [hunk 7]
```diff
@@ -1,7 +1,7 @@
 
-SERVICE_NAME=desktop-frontend
+SERVICE_NAME=zhujingyang/desktop-frontend
```
Comment (zzjin) [inline frontend/desktop/Makefile:2]: Do Not push image name test change.

File: frontend/desktop/src/components/notification/index.tsx [hunk 8]
```diff
@@ -117,22 +119,22 @@ export default function Notification(props: TNotification) {
...
-                未读 ({unread_notes?.length || 0})
+                {t('unread')} ({unread_notes?.length || 0})
               </Box>
               <Box
                 ml={'12px'}
                 className={clsx(activeTab === 'read' && styles.active, styles.tab)}
                 onClick={() => setActiveTab('read')}
               >
-                已读
+                {t('Have read')}
```
Comment (zzjin) [inline frontend/desktop/src/components/notification/index.tsx:129]: mix usage of uppercase and lowercase.

File: frontend/desktop/src/components/desktop_content/time.tsx [hunk 9]
```diff
@@ -14,6 +15,8 @@ const WeekDay = {
 
 export default function TimeComponent(props: any) {
   const [time, setTime] = useState(new Date().toString());
+  const { i18n } = useTranslation();
+  console.log(i18n.language);
```
Comment (zzjin) [inline frontend/desktop/src/components/desktop_content/time.tsx:19]: remove debug log.

**Review by zzjin — APPROVED** [review]

**Pull request merged**